src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -175,13 +270,53 @@
     private static final Pattern JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN =
             Pattern.compile("\n[ ]+(\\*)|^[ ]+(\\*)");
 
+    /**
+     * This regexp is used to remove html tags from string.
+     */
+    private static final Pattern HTML_TAG_PATTERN = Pattern.compile("<[^>]*>");
     /** Period literal. */
     private static final String PERIOD = ".";
 
+    /** Summary tag text. */
+    private static final String SUMMARY_TEXT = "@summary";
+
     /** Set of allowed Tokens tags in summary java doc. */
     private static final Set<Integer> ALLOWED_TYPES = Collections.unmodifiableSet(
             new HashSet<>(Arrays.asList(JavadocTokenTypes.TEXT,
-                    JavadocTokenTypes.WS))
+                    JavadocTokenTypes.WS,
+                    JavadocTokenTypes.DESCRIPTION,
+                    JavadocTokenTypes.TEXT))
+    );
+
+    /**
+     * Set of allowed inline tags in summary java doc.
+     *
+     * <p>These tags are allowed with respect to
+     * https://docs.oracle.com/javase/11/docs/specs/doc-comment-spec.html#general-syntax .
+     * According to which all the HTML tags and inline tags described in it are allowed.
+     * "@inheritDoc" is not included as it is forbidden by check.</p>
+     *
+     */
+    private static final Set<String> ALLOWED_INLINE_TAGS = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    "@code",
+                    "@link",
+                    "@input",
+                    "@docRoot",
+                    "@index",
+                    "@linkplain",
+                    "@literal",
+                    "@value"
+            ))
+    );
+
+    /**
+     * Set of tags that define the html tag is of type list.
+     */
+    private static final Set<String> HTML_LIST_TAGS = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    "ul", "ol", "dl"
```

Comment:
why we should care about this ?
there could be any HTML tags

Reply:
Sir, these HTML tags are unique as we need to go one step inside for these tag to extract the text inside them and verify that it ends with `PERIOD`

Reply:
Refracted now. No more checking required.

Reply:
@AkMo3 , is because of https://github.com/checkstyle/checkstyle/pull/9174#discussion_r604884666 problem ?

please update javadoc of this field to explain reason of it, it is very weird, next contributors need to know such nuances.

Reply:
Sir, this will always be required. After #9703 is resolved, we will no longer need https://github.com/checkstyle/checkstyle/blob/84a9b54752f7c7a82d508981be52fbb3fb87dcf9/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L631 , as we HTML tags will be parsed perfectly, and they would not be included in text. I will update javadoc of this method, and mention this issue and reason.

Reply:
please make link to this issue to javadoc of this field.

Reply:
Done

Reply:
I do not see link to Github issue until which it is reuqired.

Reply:
https://github.com/checkstyle/checkstyle/blob/3cff8233650af2aaea532ab93b55963fe0517594/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L644-L655

I have added this to javadoc of `removeHtmlTagsFromString()` as this is the method need needs to be amended after fix of issue.